tests/system.py

```diff
@@ -731,30 +717,29 @@ def test_editor_role_can_create_resources(
         editor_subscriber.create_subscription(subscription_path, topic_path)
         editor_subscriber.create_snapshot(snapshot_path, subscription_path)
 
-    def _publish_messages(self, publisher, topic_path, batch_sizes):
-        """Publish ``count`` messages in batches and wait until completion."""
-        publish_futures = []
-        msg_counter = itertools.count(start=1)
 
-        for batch_size in batch_sizes:
-            msg_batch = self._make_messages(count=batch_size)
-            for msg in msg_batch:
-                future = publisher.publish(
-                    topic_path, msg, seq_num=str(next(msg_counter))
-                )
-                publish_futures.append(future)
-            time.sleep(0.1)
+def _publish_messages(publisher, topic_path, batch_sizes):
```

Comment:
nit: would be nice to have helper functions at the top, instead of at the bottom

Reply:
I actually prefer the opposite, i.e. having the higher-level code at the top, and the callees topologically sorted towards the bottom, makes it easier to  follow the code in a drill-down fashion (IMO at least). One can then quickly check the high-level tests, without having to first scroll through their helpers ("implementation details"), thus I would keep the existing structure of the file.

Reply:
Okay, that's fine.